src/tabs/__tests__/tabs.test.tsx

```diff
@@ -752,6 +796,272 @@ describe('Tabs', () => {
+
+        fireEvent.mouseEnter(wrapper.findTabLinkById('second')!.getElement());
+
+        expect(wrapper.findTabLinkById('second')!.findDisabledReason()!.getElement()).toHaveTextContent(
+          'disabled reason'
+        );
+
+        fireEvent.mouseLeave(wrapper.findTabLinkById('second')!.getElement());
+
+        expect(wrapper.findTabLinkById('second')!.findDisabledReason()).toBe(null);
+      });
+
+      test('has no aria-describedby by default', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(<Tabs tabs={defaultTabs} activeTabId={firstTabId} onChange={() => void 0} />);
+
+        expect(wrapper.findTabLinkById('second')!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has no aria-describedby without disabledReason', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(
+          <Tabs
+            tabs={defaultTabs.map(item => {
+              if (item.id === 'second') {
+                return {
+                  ...item,
+                  disabled: true,
+                };
+              }
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        expect(wrapper.findTabLinkById('second')!.getElement()).not.toHaveAttribute('aria-describedby');
+      });
+
+      test('has aria-describedby with disabledReason', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(
+          <Tabs
+            tabs={defaultTabs.map(item => {
+              if (item.id === 'second') {
+                return {
+                  ...item,
+                  disabled: true,
+                  disabledReason: 'disabled reason',
+                };
+              }
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        expect(wrapper.findTabLinkById('second')!.getElement()).toHaveAttribute('aria-describedby');
+      });
+
+      test('has hidden element (linked to aria-describedby) with disabledReason', () => {
+        const firstTabId = defaultTabs[0].id;
+        const { wrapper } = renderTabs(
+          <Tabs
+            tabs={defaultTabs.map(item => {
+              if (item.id === 'second') {
+                return {
+                  ...item,
+                  disabled: true,
+                  disabledReason: 'disabled reason',
+                };
+              }
+
+              return item;
+            })}
+            activeTabId={firstTabId}
+            onChange={() => void 0}
+          />
+        );
+
+        expect(wrapper.findTabLinkById('second')!.find('span[hidden]')!.getElement()).toHaveTextContent(
+          'disabled reason'
+        );
+      });
+
+      test('show previously selected tab content when disabled with reason tab is focused', () => {
```

Comment:
nit: I'd phrase it as "shows active tab content when disabled with reason tab is focused" (it if actually works that way). Otherwise, the previously selected tab can be another tab with disabled reason.

Reply:
Updated